traffic_portal/test/integration/specs/Login.spec.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { browser } from 'protractor';
+import { LoginPage } from '../PageObjects/LoginPage.po';
+import { TopNavigationPage } from '../PageObjects/TopNavigationPage.po';
+import { API } from '../CommonUtils/API';
+import { isMainThread } from 'worker_threads';
+
+let fs = require('fs')
+let using = require('jasmine-data-provider');
+let filename = 'Data/Login/TestCases.json';
+let testData = JSON.parse(fs.readFileSync(filename));
+
+let api = new API();
+let topNavigation = new TopNavigationPage();
+let loginPage = new LoginPage();
+
+using(testData.LoginTest, async function(loginData){
+    using(loginData.Login, function(login){
+        describe('Traffic Portal - Login - '+ login.description, function(){
+            it('can open login page', async function(){
+                browser.get(browser.params.baseUrl);
+            })
+            it(login.description, async function(){
+                expect(await loginPage.Login(login)).toBeTruthy();
+            })
+        })
+    })
+})
```

Comment:
Missing newline at EOF

Reply:
> Missing newline at EOF

do we need some sort of linter that runs via a GHA to catch this sort of thing?

Reply:
I think we'd need to write our own linter, since that should apply to every text file in the repository, independent of framework, language or project. It's not a bad idea, just might be more work than it immediately sounds.

Reply:
Added new line at EOF

Reply:
well, we do have @ocket8888 - he's our linter for now :)